mythicmaker/src/components/ProtectedRoute.tsx

```diff
@@ -0,0 +1,16 @@
+// ProtectedRoute.js
+import React from 'react';
+import { Navigate } from 'react-router-dom';
+import { useAuth } from './../AuthProvider';
+
+const ProtectedRoute = ({ component: Component, restricted, ...rest }) => {
+  const { currentUser } = useAuth();
+
+  return currentUser && restricted ? (
+    <Navigate to="/character" />
+  ) : (
+    <Component {...rest} />
+  );
+};
```

Comment:
To powinno być tak, że jeżeli jest restricted ORAZ nie ma currentUsera, to przenosisz usera na login page, bo rozumiem ze strona jest zabezpieczona ale nie jestes zalogowany.

Jezeli jest restricted i current user LUB not restricted to wtedy component rest

Ma to sens?